ckan/views/admin.py

```diff
@@ -169,6 +168,36 @@ def __init__(self):
             }
         }
 
+    def _get_deleted_datasets(
+        self
+    ) -> Union["Query[model.Package]", List[Any]]:
+        if config.get_value('ckan.search.remove_deleted_packages'):
+            return self._get_deleted_datasets_from_db()
+        else:
+            return self._get_deleted_datasets_from_search_index()
+
+    def _get_deleted_datasets_from_db(self) -> "Query[model.Package]":
+        return model.Session.query(
+            model.Package
+        ).filter_by(
+            state=model.State.DELETED
+        )
+
+    def _get_deleted_datasets_from_search_index(self) -> List[Any]:
+        query = search.query_for(model.Package)
+        search_params = {
+            'fq': '+state:deleted',
+            'df': 'text',
+            'fl': 'id name title dataset_type',
+        }
+        query.run(search_params)
```

Comment:
What you did is what I meant @avdata99 

Incidentally, ` package_search({"include_deleted": True, "fq": "+state:deleted"})` works when called directly from python, but it doesn't when coming from an HTTP request to the API, because of how params and parsed and the `fq` checks performed. But that's something for another PR